### DISCUSSION TITLE
How to upload a file into a particular folder  and not in the root with REST API?

### DISCUSSION THREAD

**Discussion by biapar** [directus/directus#10130] [Archive]
How to upload with REST API a file into a particular folder  and not in the root?
Thanks

**Comment by rijkvanzanten** [comment]
From https://docs.directus.io/reference/files/#upload-a-file :

> The file contents has to be provided in a part called file. All other properties of the file object can be provided as parts as well.
> 
> Order Matters
> Make sure to define the non-file properties first. This ensures that the file metadata is associated with the correct file.

For example (also from that page):

```js
import axios from 'axios';

const fileInput = document.querySelector('input[type="file"]');
const formData = new FormData();

formData.append('folder', '35f91753-254b-49ec-8e80-61bd824775dd');
formData.append('file', fileInput.files[0]);

await axios.post('/files', formData);
```

**Reply by krazyjakee** [reply]
For anyone who got here and it still isn't working, it matters which order these are in.
Make sure you append folder first, then file.

**Reply by rijkvanzanten** [reply]
As per the comment in the docs and above:

> Order Matters
Make sure to define the non-file properties first. This ensures that the file metadata is associated with the correct file.

😄

**Reply by krazyjakee** [reply]
@rijkvanzanten might be worth including the folder property in the example code.

**Reply by rijkvanzanten** [reply]
@krazyjakee Good one, sounds like something more people would try 👍🏻

**Reply by sookoll** [reply]
@rijkvanzanten what about utilsExport? There is also file prop, but I'm not able to get it working. My issue is described here: https://github.com/directus/directus/discussions/22460

So basically what I want, is to use directus SDK api to let my app user to download set of items as CSV file and I haven't found easy way to do it. As utilsExport do not return file data that it created. So I thought that if I tell directus to create new folder on every export and put that file into it, I can query that folder content. But even that do not work. Creating folder with SDK is successful as I get response with new folder name, parent I provide and id, but no such folder is visible in directus admin site. And when I try to use utilsExport with that folder, it still will create file into root.

```javascript
const folder = await directus.request(createFolder({
  name: 'random_folder_name',
  parent: 'existing_parent_folder_id'
}))

await directus.request(utilsExport(
  'collection_name',
  'csv',
  {
    filter: {
      status: {
        _eq: 'published',
      },
    },
  },
  {
    file: {
      folder: folder.id,
    },
  }
));
```
